app/scripts/components/analysis/page-hero-media.tsx

```diff
@@ -26,18 +28,45 @@ const mapOptions: Partial<mapboxgl.MapboxOptions> = {
 
 interface PageHeroMediaProps {
   feature: Feature<MultiPolygon>;
+  isHeaderStuck: boolean;
 }
 
 function PageHeroMedia(props: PageHeroMediaProps) {
-  const { feature, ...rest } = props;
+  const { feature, isHeaderStuck, ...rest } = props;
   const mapContainer = useRef<HTMLDivElement>(null);
   const mapRef = useRef<mapboxgl.Map>(null);
   const [isMapLoaded, setMapLoaded] = useState(false);
 
   const theme = useTheme();
 
+  // Delay the mount of the hero media to avoid the map showing with the wrong
+  // size. See: https://github.com/NASA-IMPACT/veda-ui/issues/330
+  // The issue:
+  // If on the analysis define page we press the generate button when the header
+  // is stuck (this happens if we scrolled down the page) then the analysis
+  // results page will be loaded with the header also stuck. This causes the map
+  // to mount and read the parent height when it is contracted. The header will
+  // then expand but the map will not.
+  // The solution:
+  // If the header starts as stuck we delay the mounting of the map so by the
+  // time the map mounts, the parent already has its final size. In the case
+  // that the header is already unstuck when the page loads, this is not needed.
+  const [shouldMount, setShouldMount] = useState(!isHeaderStuck);
+  useEffectPrevious(
+    ([wasStuck]) => {
+      if (!shouldMount && wasStuck && !isHeaderStuck) {
+        const tid = setTimeout(() => setShouldMount(true), 350);
```

Comment:
If I understood correctly this matches the animation duration defined [here](https://github.com/NASA-IMPACT/veda-ui/pull/375/files#diff-9fd873109ee581a07519c6310070b2505bcc1ad2df4e4d5dc7cec9da890d04f5L75-L76). Try to store this in a variable and reuse?